src/autocomplete/PeopleAutocomplete.js

```diff
@@ -32,6 +32,12 @@ class PeopleAutocomplete extends PureComponent<Props> {
     const { users, onAutocomplete } = this.props;
     const user = users.find(x => x.email === email);
     if (user) {
+      // If another user with the same full name is found, we send the
+      // user ID as well, to ensure the mentioned user is uniquely identified.
+      if (users.find(x => x.full_name === user.full_name && x.user_id !== user.user_id)) {
```

Comment:
I expect this will look different when https://github.com/zulip/zulip-mobile/issues/3339 is resolved, but it looks good to me as a fix before then.